mentat/code_context.py

```diff
@@ -232,59 +185,60 @@ async def _get_code_message(
         model: str,
         max_tokens: int,
     ) -> str:
-        session_context = SESSION_CONTEXT.get()
-        config = session_context.config
+        ctx = SESSION_CONTEXT.get()
 
-        code_message = list[str]()
+        code_message: List[str] = []
 
-        self.diff_context.clear_cache()
+        if self.diff_context:
+            self.diff_context.clear_cache()
         self.set_code_map()
-        if self.diff_context.files:
+        if self.diff_context and self.diff_context.files:
             code_message += [
                 "Diff References:",
                 f' "-" = {self.diff_context.name}',
                 ' "+" = Active Changes',
                 "",
             ]
         code_message += ["Code Files:\n"]
-
         features = self._get_include_features()
-        meta_tokens = count_tokens("\n".join(code_message), model)
+        meta_tokens = count_tokens("\n".join(code_message), model)  # NOTE: why does this take so long to run?
```

Comment:
Are you sure this is the bottle neck? I don't notice much latency when I put print statements around it.

Reply:
Anecdotal but I this hung for like 5-6 seconds when I was testing things a while ago. Out of scope for this PR anyways but will look into later

Reply:
I'm digging into code_context performance now. I'm seeing this line takes 0.15 seconds the first time, but if I call it immediately after it only takes .00004 seconds, so I guess it has to do with loading `tiktoken` the first time.